core/src/ast/types.rs

```diff
@@ -575,6 +595,22 @@ impl TypeName {
                     } else {
                         panic!("Expected angle brackets for Result type")
                     }
+                } else if is_runtime_type(p, "DiplomatResult") {
```

Comment:
nit: i would like these two branches to be merged

(it's a bit cleaner if the two enum variants are one variant)